syntax = "proto3";

package save_notes_service;


message SaveNotesRequest {
  string user_id = 1; 
  repeated string file_ids = 2; 
}

message SaveNotesResponse {
  repeated string saved_file_ids = 1; 
  bool success = 2; 
}

message DeleteSavedNoteRequest {
  string user_id = 1; 
  string file_id = 2; 
}

message DeleteSavedNoteResponse {
  bool success = 1; 
}

service SaveNotes {
  rpc SaveNotes (SaveNotesRequest) returns (SaveNotesResponse);

  rpc DeleteSavedNote (DeleteSavedNoteRequest) returns (DeleteSavedNoteResponse);
}
